tools/coqdep/lib/common.ml

```diff
@@ -320,6 +320,40 @@ let sort st =
   in
   List.iter (fun (name, _) -> loop name) !vAccu
 
+let expand_wildcards paths =
+  let wild_re = Str.regexp {|[\*\?\[]|} in
+  let rec split paths wild nonwild =
+    match paths with
+    | path :: tl ->
+      (try ignore(Str.search_forward wild_re path 0); split tl (path :: wild) nonwild
+      with Not_found -> split tl wild (path :: nonwild))
+    | [] -> List.rev wild, List.rev nonwild
+  in
+  let open CoqProject_file in
+  let paths = List.map (fun {thing} -> thing) paths in
+  let (wild, nonwild) = split paths [] [] in
+  let temp_make = Filename.temp_file "coqdep" ".make" in
+  let fout = open_out temp_make in
+  let makefile = Printf.sprintf "X = $(wildcard %s)\n\n\
+                          all:\n\
+                          \t@for name in $X; do \\\n\
+                          \t\techo $${name}; \\\n\
+                          \tdone\n" (String.concat " " wild) in
+  Printf.fprintf fout "%s" makefile;
+  close_out fout;
+  let temp = Filename.temp_file "coqdep" "wild" in
+  let cmd = Printf.sprintf "make -j 1 --no-print-directory -f %s >%s" temp_make temp in
```

Comment:
I cannot agree with such a code. At the very least, do not call `make` if there are no wildcards.

Reply:
It's unconditional here to debug a problem.